cli/src/cpp/utils/external_locators_parser.cpp

```diff
@@ -122,31 +128,21 @@ void external_locators_parser(
         }
     }
 
-    if (!print_usage && (!subelement.empty() || set_every_element))
+    if (!subelement.empty() || set_every_element)
     {
         if (!subelement.empty())
         {
             std::string dummy_subelement;
             std::string dummy_key;
-            // Subelement should not be keyed
-            if (extract_element_subelement_key(subelement, dummy_subelement, dummy_key))
-            {
-                std::cout << "ERROR: external locator <" << subelement << "> attribute is not keyed" << std::endl;
-                print_usage = true;
-            }
-            // Subelement should be final
-            else if (!dummy_subelement.empty())
-            {
-                std::cout << "ERROR: external locator <" << subelement << "> attribute is FINAL element" << std::endl;
-                print_usage = true;
-            }
+            keyed = extract_element_subelement_key(subelement, dummy_subelement, dummy_key);
+            message << " <" << subelement << "> attribute";
+            // Should not be keyed
+            print_usage = print_usage || !check_keyed(false, keyed, message.str());
+            // Should be final
+            print_usage = print_usage || !check_final_element(true, dummy_subelement, message.str());
             // SET command requires only one argument
-            else if (values.size() != 1 && command == CommonCommands::SET)
-            {
-                std::cout << "ERROR: external locator <" << element << "[]." << subelement <<
-                    "> requires only 1 configuration parameter" << std::endl;
-                print_usage = true;
-            }
+            print_usage = print_usage || (!check_command_arguments(command, 1, values.size(), message.str(), true) &&
+                    CommonCommands::SET == command);
```

Comment:
NIT Just for uniformity

```suggestion
            print_usage = print_usage || (CommonCommands::SET == command 
 && !check_command_arguments(command, 1, values.size(), message.str(), true));
```

Reply:
🤦